ladon.go

```diff
@@ -48,13 +56,23 @@ func (l *Ladon) IsAllowed(r *Request) (err error) {
 // The IsAllowed interface should be preferred since it uses the manager directly. This is a lower level interface for when you don't want to use the ladon manager.
 func (l *Ladon) DoPoliciesAllow(r *Request, policies []Policy) (err error) {
 	var allowed = false
+	var deciders = Policies{}
+
+	defer func() {
+		if allowed {
+			l.auditLogger().OnGrantedAuthorization(r, policies, deciders)
+		} else {
+			l.auditLogger().OnDeniedAuthorization(r, policies, deciders)
+		}
+	}()
 
 	// Iterate through all policies
 	for _, p := range policies {
 		// Does the action match with one of the policies?
 		// This is the first check because usually actions are a superset of get|update|delete|set
 		// and thus match faster.
 		if pm, err := l.matcher().Matches(p, p.GetActions(), r.Action); err != nil {
+			deciders = append(deciders, p)
```

Comment:
Not sure about appending these here, we are explicitly experiencing an error (e.g. invalid regular expression), in that case, the app logic should handle the error appropriately.